File: typescript/src/v2/chromatic_circle/mixers/YellowMixer.ts
import { Color } from '../Color';
import { ColorMixer } from './ColorMixer';

export class YellowMixer implements ColorMixer {
  private readonly color: Color;

  constructor() {
    this.color = Color.YELLOW;
  }

  public canMix(color: Color): boolean {
    return color === this.color;
  }

  public mixColors(color: Color): Color {
    if (color == this.color) {
      return color;
    }
    if (color === Color.RED) {
      return Color.ORANGE;
    }
    if (color === Color.BLUE) {
      return Color.GREEN;
    }
    throw new Error(`invalid color: ${color}`);
  }
}

File: java/src/main/java/com/circulocromatico/YellowMixer.java
package com.circulocromatico;

public class YellowMixer implements ColorMixer {
    public static final ChromaticCircle.Color YELLOW = ChromaticCircle.Color.YELLOW;

    @Override
    public boolean canMix(ChromaticCircle.Color color) {
        return color == YELLOW;
    }

    @Override
    public ChromaticCircle.Color mixColor(ChromaticCircle.Color color) {
        if (YELLOW.equals(color)) {
            return color;
        }
        if (color == ChromaticCircle.Color.RED) {
            return ChromaticCircle.Color.ORANGE;
        }
        if (color == ChromaticCircle.Color.BLUE) {
            return ChromaticCircle.Color.GREEN;
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }
}